src/wix/ecom/session.ts

```diff
@@ -35,31 +35,57 @@ export { commitSession, destroySession, getSession };
 
 export async function initializeEcomSession(request: Request) {
     const session = await getSession(request.headers.get('Cookie'));
-
     const sessionWixClientId = session.get('wixClientId');
     const wixClientId = getWixClientId();
 
     // Retrieve session tokens only if the OAuth client ID matches the one associated with the session;
     // otherwise, reset the session tokens due to client ID mismatch.
-    let wixSessionTokens =
+    const wixSessionTokens =
         sessionWixClientId === wixClientId ? session.get('wixSessionTokens') : undefined;
-    let shouldUpdateSessionCookie = false;
 
-    const client = createWixClient(wixSessionTokens);
-    if (wixSessionTokens === undefined) {
+    const effectiveTokens = await getEffectiveWixAuthTokens(wixSessionTokens);
```

Comment:
what happens if token gets expired during the session?

Reply:
see comment here: https://github.com/codux-templates/e-commerce-reclaim/pull/242#discussion_r1842035310